runtime/lua/vim/lsp/buf.lua

```diff
@@ -87,6 +101,33 @@ function M.hover(config)
       else
         vim.list_extend(contents, util.convert_input_to_markdown_lines(result.contents))
       end
+      local range = result.range
+      if range then
+        local start_line = range.start.line
+        local start_char = range.start.character
+        local end_line = range['end'].line
+        local end_char = range['end'].character
+
+        local start_idx = util._get_line_byte_from_position(
+          bufnr,
+          { line = start_line, character = start_char },
+          client.offset_encoding
+        )
+        local end_idx = util._get_line_byte_from_position(
+          bufnr,
+          { line = start_line, character = end_char },
+          client.offset_encoding
+        )
+
+        vim.hl.range(
+          bufnr,
+          hover_ns,
+          'LspHoverRange',
+          { start_line, start_idx },
+          { end_line, end_idx },
+          { priority = vim.hl.priorities.user }
+        )
+      end
```

Comment:
```suggestion
      if range then
        local start = range.start
        local end_ = range["end"]
        local start_idx = util._get_line_byte_from_position(
          bufnr,
          start,
          client.offset_encoding
        )
        local end_idx = util._get_line_byte_from_position(
          bufnr,
          end_,
          client.offset_encoding
        )

        vim.hl.range(
          bufnr,
          hover_ns,
          'LspHoverRange',
          { start.line, start_idx },
          { end_.line, end_idx },
          { priority = vim.hl.priorities.user }
        )
      end
```

No need to unpack the start/end position only to create a new table in the same format.